File: app/src/main/java/tech/salroid/filmy/Network/VolleySingleton.java
package tech.salroid.filmy.network;

import com.android.volley.RequestQueue;
import com.android.volley.toolbox.Volley;

import tech.salroid.filmy.FilmyApplication;

/**
 * Created by Home on 7/23/2016.
 */
public class VolleySingleton {

    public static VolleySingleton instance = null;
    private RequestQueue requestQueue;


    private VolleySingleton() {
        requestQueue = Volley.newRequestQueue(FilmyApplication.getContext(), new MyHurlStack());
    }


    public static VolleySingleton getInstance() {

        if (instance == null) {
            instance = new VolleySingleton();
        }


        return instance;
    }


    public RequestQueue getRequestQueue() {
        return requestQueue;
    }
}

File: app/src/main/java/tech/salroid/filmy/networking/VolleySingleton.kt
package tech.salroid.filmy.networking

import tech.salroid.filmy.FilmyApplication.Companion.context
import com.android.volley.RequestQueue
import com.android.volley.toolbox.Volley

object VolleySingleton {
    val requestQueue: RequestQueue = Volley.newRequestQueue(context, MyHurlStack())
}